### PR TITLE
Windows: Updated the create manifest script to support multiple registries

### PR DESCRIPTION
**Pull request by rawahars**
## Summary
Presently, we are publishing the manifest of images to `regional ECR`, `Public ECR`, and `DockerHub`. Additionally, we also have a workflow to sync stable image in each region, wherein we create the manifest for the stable image.

Therefore, in this change, we are-
- Creating a common script which has common methods across various Windows scripts which we would be using in the release infrastructure.
    - Methods to set the required variables such as `All supported versions`, `Latest version`, `Stable version`, etc.
    - Method to check if the image manifest for Windows exists in a region.
    - Method to check if the two manifests are same.
- Refactor the existing create manifest script to use multiple registries.
- Changed the CMD in Windows dockerfile to exec format.
- Created a new file for Windows stable version.
- Updated versions in windows.version

## Testing
These scripts have been tested as part of an automated end to end workflow.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: scripts/common_windows.sh [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+# language governing permissions and limitations under the License.
+
+set -uo pipefail
+
+#############################################################################################################
+# Helper methods to set the required variables.
+#############################################################################################################
+# Constant
+# Number of presently supported OS versions.
+SUPPORTED_WINDOWS_OS_RELEASES_NUMBER=2
```
Comment (jterry75) [inline scripts/common_windows.sh:22]: What does this mean?

File: scripts/common_windows.sh [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    if [[ "$AWS_FOR_FLUENT_BIT_LATEST_VERSION" < "$version" ]]; then
+      AWS_FOR_FLUENT_BIT_LATEST_VERSION=$version
+    fi
+  done <<< "$(echo "$ALL_AWS_FOR_FLUENT_BIT_VERSIONS")"
+}
+
+# Sets AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB variable with the latest version available in dockerhub.
+set_dockerhub_version() {
+  # Find the dockerhub version.
+  docker_hub_image_tags=$(curl -s -S 'https://registry.hub.docker.com/v2/repositories/amazon/aws-for-fluent-bit/tags/?page=1&page_size=250' | jq -r '.results[].name')
```
Comment (jterry75) [inline scripts/common_windows.sh:49]: When we push to dockerhub are they full layers or are they foreign layers?

File: scripts/common_windows.sh [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  if [ "${first_manifest}" = 1 ]; then
+    echo "No manifest found for ${first_manifest_image}"
+    return 1
+  elif [ "${second_manifest}" = 1 ]; then
+    echo "No manifest found for ${second_manifest_image}"
+    return 1
+  fi
+
+  # Obtain the digest entries from the manifest.
+  first_manifest_digests=$(echo "${first_manifest}" | jq -r '.manifests[].digest' | sort)
```
Comment (jterry75) [inline scripts/common_windows.sh:111]: This wont work on a manifest list right?

File: scripts/common_windows.sh (hunk 1 above)
Comment (rawahars) [reply scripts/common_windows.sh:22]: So this is essentially the count of number of supported Windows releases. Right now it is 2 i.e. Windows Server 2019 and Windows Server 2022.

In `check_image_manifest_exists` method, we check if the image manifest for the given version does exist in the target registry and the number of digests is equal to this number i.e. 2.

File: scripts/common_windows.sh (hunk 3 above)
Comment (rawahars) [reply scripts/common_windows.sh:111]: So this method does compare manifest lists. `$(echo "${first_manifest}" | jq -r '.manifests[].digest' | sort)` returns a list of digests which we compare for both the images.

File: scripts/common_windows.sh (hunk 2 above)
Comment (rawahars) [reply scripts/common_windows.sh:49]: So as discussed offline, currently we are pushing full layers to dockerhub.

**Review by jterry75 — APPROVED** [review]

**Review by PettitWesley — APPROVED** [review]

**Pull request merged**